### ISSUE TITLE
Add ability to output JS / non-html files

### ISSUE THREAD

**Issue by prencher** [jantimon/html-webpack-plugin#696]
## Description

We have need of generating a templated index.js for our builds. Currently, this works if the file extension is not actually `.js`. However, if the filename is changed to `index.js`, the following happens:

```
ERROR in   Error: The loader "/Users/thomas/projects/dasher/node_modules/html-webpack-plugin/lib/loader.js!/Users/thomas/projects/dasher/src/core/bootstrap/index.js.ejs" didn't return   html.
```

**Comment by mastilver** [comment]
What do you means by: "generating a templated index.js for our builds."?

This plugin will only output html!

**Comment by prencher** [comment]
It will output whatever your template outputs, as long as you turn inject off. 😄 

The problem is that it gives the above error if the output filename doesn't end with html - it completely works otherwise.

If we could allow that to happen, it would make the plugin useful for a slew of other use cases. My cursory look into the codebase didn't reveal any obvious way for me to do it myself and submit a PR, however.

**Comment by mastilver** [comment]
> It will output whatever your template outputs, as long as you turn inject off. 😄

I trust you on this, though it's not the purpose of this plugin, I will try to help you

Can you give me your loaders or even better you entire `webpack.config.js`

**Comment by jantimon** [comment]
@prencher could you please try to trace the error back to our code?

**Comment by prencher** [comment]
So sorry for delay on responding here, ran into this and then went on vacation shortly after. Here's the plugin configuration:
```js
new HtmlWebpackPlugin({
  filename: env.embedded ? 'index.js' : 'index.html',
  template: path.join(__dirname, 'src/core/bootstrap', env.embedded ? 'index.js.ejs' : 'index.html.ejs'),
  inject: false,
  minify: !env.embedded && env.optimized && {
    minifyCSS: true,
    minifyJS: true,
    collapseWhitespace: true,
  },
}),
```

**Comment by martinek** [comment]
Is there any progress with this feature?

**Comment by robertaistleitner** [comment]
I tried to find the root cause of the issue but I failed. It looks like the output extension triggers compilation of the generated javascript a second time which then fails because nothing is returned any longer. I tried to exclude the target filename from the rules that do the javascript compilation but somehow it still gets processed.

If I choose any other extension (like `.aaa`), it works..

**Comment by tubalmartin** [comment]
Same issue here.

**Comment by robertaistleitner** [comment]
I have an update on this one. I found a solution some time ago but forgot to keep you posted here and just stumbled upon this issue again, so here's how I worked around it.

The root problem was that the `UglifyJsPlugin` was processing the generated javascript file again, so what I did was to make sure that it's excluded from processing by configuring it like this:

```javascript
new webpack.optimize.UglifyJsPlugin({
  exclude: [/loader\.js$/],
}),
```
where `loader.js` is the filename of the script I use as the template.

If you're using `webpack -p` to build your project for production, make sure you remove it and configure `UglifyJsPlugin` manually!
